src/fuzzer.cpp

```diff
@@ -0,0 +1,385 @@
+#include <bits/stdc++.h>
+#include "rmm_tree.h"
+#include "naive_rmm_tree.h"
+
+using std::size_t;
+static constexpr size_t NPOS = std::numeric_limits<size_t>::max();
+
+static std::string bits_to_parens(const std::string &bits)
+{
+    std::string s;
+    s.reserve(bits.size());
+    for (char c : bits)
+        s.push_back(c == '1' ? '(' : ')');
+    return s;
+}
+static std::string vecbits_to_string(const std::vector<uint8_t> &v)
+{
+    std::string s;
+    s.resize(v.size());
+    for (size_t i = 0; i < v.size(); ++i)
+        s[i] = v[i] ? '1' : '0';
+    return s;
+}
+
+static std::string random_bits(std::mt19937_64 &rng, size_t n)
+{
+    std::uniform_int_distribution<int> b01(0, 1);
+    std::string s;
+    s.resize(n);
+    for (size_t i = 0; i < n; i++)
+        s[i] = char('0' + b01(rng));
+    return s;
+}
+
+static std::string random_dyck_bits(std::mt19937_64 &rng, size_t m)
+{
+    std::string s;
+    s.resize(2 * m);
+    std::bernoulli_distribution coin(0.5);
+    int h = 0;
+    for (size_t pos = 0; pos < 2 * m; ++pos)
+    {
+        size_t rem = 2 * m - pos;
+        if (h == 0)
+        {
+            s[pos] = '1';
+            ++h;
+            continue;
+        }
+        if ((size_t)h == rem - 1)
+        {
+            s[pos] = '0';
+            --h;
+            continue;
+        }
+        s[pos] = coin(rng) ? '1' : '0';
+        if (s[pos] == '1')
+            ++h;
+        else
+            --h;
+        if (h < 0)
+        {
+            s[pos] = '1';
+            h += 2;
+        }
+    }
+
+    if (h > 0)
+    {
+        for (size_t i = s.size(); i > 0 && h > 0; --i)
+        {
+            if (s[i - 1] == '1')
+            {
+                s[i - 1] = '0';
+                --h;
+            }
+        }
+    }
+    return s;
+}
+
+template <class T>
+static std::string show_sz_or_npos(T x)
+{
+    if constexpr (std::is_same_v<T, size_t>)
+        return (x == NPOS ? std::string("npos") : std::to_string(x));
+    else
+        return std::to_string(x);
+}
+
+[[noreturn]] static void die_mismatch(
+    const std::string &seed_info,
+    const std::string &bits,
+    const char *op_name,
+    const std::vector<std::string> &args,
+    const std::string &want,
+    const std::string &got)
+{
+    std::cerr << "\n=== MISMATCH DETECTED ===\n";
+    std::cerr << seed_info << "\n";
+    std::cerr << "bits:   " << bits << "\n";
+    std::cerr << "parens: " << bits_to_parens(bits) << "\n";
+    std::cerr << "op: " << op_name << "(";
+    for (size_t i = 0; i < args.size(); ++i)
+    {
+        if (i)
+            std::cerr << ", ";
+        std::cerr << args[i];
+    }
+    std::cerr << ")\n";
+    std::cerr << "expected: " << want << "\n";
+    std::cerr << "got:      " << got << "\n";
+    std::cerr << "=========================\n";
+    std::exit(1);
+}
+
+int main()
```

Comment:
Unless benchmarks our imported from some third_party code "as is" please use google benchmark for benchmarking